uro98/assignment3/FindAllWordsTest.java

```diff
@@ -0,0 +1,66 @@
+import static org.junit.Assert.*;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.junit.Test;
+
+public class FindAllWordsTest {
```

Comment:
It would be great to add tests for edge cases too. What would be some other good scenarios to check to make sure that findWords returns only words that exist in the dictionary and only ones that can actually be constructed from the grid?

Reply:
Apart from empty grids and dictionaries, doesn't the first test cover both of those scenarios?

Reply:
You are right. Checking whether returned set of words is equal to set of expected words does indeed verify that.